source/CIS/cis-4.rst

```diff
@@ -0,0 +1,750 @@
+
+  CredentialSchemaRefEvent ::= (245: Byte) (type: CredentialType) (schema_ref: SchemaRef)
+
+``RevocationKeyEvent``
+^^^^^^^^^^^^^^^^^^^^^^
+
+A ``RevocationKeyEvent`` event MUST be logged when registering a new or removing an existing revocation key.
+It consist of the key and the action performed with the key (registration or removal).
+
+The ``RevocationKeyEvent`` event is serialized as: first a byte with the value of 250, followed by the key bytes :ref:`CIS-4-PublicKeyEd25519` and 1 byte encoding the action (0 for ``Register``, 1 for ``Remove``)::
+
+  RevocationKeyAction ::= (0: Byte)    // Register
+                        | (1: Byte)    // Remove
+  RevocationKeyEvent ::= (244: Byte) (action: RevocationKeyAction)
+
+
+.. _CIS-4-issuer-metadata-json:
+
+Issuer metadata JSON
+--------------------
+
+The issuer metadata is stored off-chain and MUST be a JSON (:rfc:`8259`) file.
+
+All of the fields in the JSON file are optional, and this specification reserves a number of field names, shown in the table below.
+
+.. list-table:: Issuer metadata JSON Object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``name``
+    - string
+    - The name to display for the issuer.
+  * - ``icon``
+    - URL JSON object
+    - An image URL for displaying the issuer.
+  * - ``description`` (optional)
+    - string
+    - A description for the issuer.
+  * - ``url`` (optional)
+    - string (:rfc:`3986`) [``uri-reference``]
+    - A URL of the issuer's website.
+
+Optionally a SHA256 hash of the JSON file can be logged with the :ref:`CIS-4-events-IssuerMetadata` event for checking integrity.
+Since the metadata JSON file could contain URLs, a SHA256 hash can optionally be associated with the URL.
+To associate a hash with a URL the JSON value is an object:
+
+.. list-table:: URL JSON Object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``url``
+    - string (:rfc:`3986`) [``uri-reference``]
+    - A URL.
+  * - ``hash`` (optional)
+    - string
+    - A SHA256 hash of the URL content encoded as a hex string.
+
+Example issuer metadata
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: json
+
+  {
+    "name": "Concordium",
+    "icon" : {
+      "url":  "https://concordium.com/wp-content/uploads/2022/07/Concordium-1.png",
+      "hash": "1c74f7eb1b3343a5834e60e9a8fce277f2c7553112accd42e63fae7a09e0caf8"
+      }
+    "description": "A public-layer 1, science-backed blockchain",
+    "url": "https://concordium.com"
+  }
+
+Credential metadata JSON
+------------------------
+
+The credential metadata is stored off-chain and MUST be a JSON (:rfc:`8259`) file.
+
+.. list-table:: Credential metadata JSON Object
+  :header-rows: 1
+
+  * - Property
+    - JSON value type [JSON-Schema]
+    - Description
+  * - ``title``
+    - string
+    - The name to display for the credential.
+  * - ``logo``
```

Comment:
Do we want to specify formats (sizes) in the standard document? Or do we want to do that elsewhere?

Reply:
Good question. CIS-2 does not specify the sizes. I think it makes sense to leave them out. CIS standards are about the API, additional details might be specified elsewhere.

@orhoj is there a document that specifies the image sizes for CIS-2 complaint NFTs?

Reply:
Let's add a recommendation about the image sizes to the standard. This recommendation might be revised later depending on the final version of the wallet's UI.

Reply:
Done. I added initial values and a TODO to check before we finalize the standard.